### DISCUSSION TITLE
Controls onZoomIn/onZoomOut & docs difference clarification

### DISCUSSION THREAD

**Discussion by badsyntax** [xyflow/xyflow#4405] [Q&A] [answered]
Hi there

The [docs for <Controls /> onZoomIn](https://reactflow.dev/api-reference/components/controls#on-zoom-in) state that `When this is not provided, a default zoom will be applied to the viewpor.t`. To me this means if I provide an onZoomIn handler, the default zoom would not be applied, when infact the default zoom is ALWAYS applied,  as you can see here: https://github.com/xyflow/xyflow/blob/41cefdfb0d7575a314e5cdd6d3e60f417cc518e3/packages/react/src/additional-components/Controls/Controls.tsx#L45

I'm trying to add zoom animation to the default controls and thought I'd be able to do this:

```ts
import { Controls as BaseControls, useReactFlow } from 'reactflow';

export function Controls() {
  const { zoomIn, zoomOut } = useReactFlow();

  return (
    <BaseControls
      fitViewOptions={{ duration: 800 }}
      onZoomIn={() => {
        zoomIn({ duration: 800 });
      }}
      onZoomOut={() => {
        zoomOut({ duration: 800 });
      }}
    />
  );
}
```

The above code will call `zoomIn` twice.

So just wondering if this is a bug or if the docs can be improved or if i'm misunderstanding the docs?

**Comment by hayleigh-dot-dev** [comment, marked as answer]
Hey good catch! The docs are worded slightly incorrectly here, so clear things up the viewport is _always_ zoomed by a fixed amount **and** if an `onZoomIn` callback is provided we call that in addition.

If you want to customise the zoom behaviour you should hide the default zoom buttons entirely and render your own controls using the `<ControlButton />` component.

---

I'll make sure the docs get updated now!

**Reply by badsyntax** [reply]
Thanks!
